src/v/cluster/tests/tx_compaction_utils.h

```diff
@@ -319,11 +319,11 @@ class tx_executor {
               .last_seq = spec.count - 1,
               .record_count = spec.count,
               .is_transactional = true};
-            auto reader = model::make_memory_record_batch_reader(
-              std::move(batches[0]));
+
             auto result = co_await _ctx._stm->replicate(
               bid,
-              std::move(reader),
+              chunked_vector<model::record_batch>::single(
+                std::move(batches[0])),
```

Comment:
did you add a single batch interface to replicate so you don't need the ::single factory here?

Reply:
this is a bit different situation as `replicate` comes from `rm_stm`